### PR TITLE
feat: create component map location

### PR DESCRIPTION
**Pull request by IounOm**
## What does it do?
- feat: create component map location

## Screenshot
Screenshot 2567-10-17 at 16 48 15
Screenshot 2567-10-17 at 16 47 45

## Related issue
-

### CODE REVIEW

File: src/components/map-location.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import type { EventsSubPages } from "@/content/eventsSubPages/eventsSubPages";
+import { cn } from "@/lib/utils";
+
+const MapLocation = (
+  props: Readonly<{
+    eventsSubPages: EventsSubPages;
```
Comment (JeanneGrenet) [inline src/components/map-location.tsx:6]: This component will also be used in meetup "home" page. I think we should have `title` and `address` but not in the event subpages

**Review by JeanneGrenet — CHANGES_REQUESTED** [review]

File: src/components/map-location.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  }>,
+) => {
+  return (
+    <div className="flex h-full w-full flex-col gap-2">
+      <div className="flex flex-col">
+        <p className="my-0">{props.eventsSubPages.title}</p>
+        {props.eventsSubPages.description && (
+          <p className="my-0 text-neutral-400">
+            {props.eventsSubPages.description}
+          </p>
```
[the remark this answers is not in the document]
Comment (IounOm) [reply src/components/map-location.tsx:18]: I noticed that the map on the meetup page doesn't have a pin icon, so I didn't include it.
The next steps are:
- Add the pin icon.
- Create props to show/hide the icon for reuse on other pages.

Screenshot 2567-10-22 at 17 30 34

File: src/components/map-location.tsx (hunk 1 above)
Comment (IounOm) [reply src/components/map-location.tsx]: fixed!

File: src/components/map-location.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (IounOm) [reply src/components/map-location.tsx]: fixed!

[the remark this answers is not in the document]
Comment (IounOm) [reply src/components/map-location.tsx]: fixed!

File: src/components/map-location.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    hideIcon?: boolean;
+    mapUri: string;
+    className?: string;
+  }>,
+) => {
+  const { hideIcon = false, mapUri, className } = props;
+  return (
+    <div className="flex h-full w-full flex-col gap-2">
+      <div className="flex gap-1">
+        {!hideIcon && <div className="pt-2">{ICONS.pin}</div>}
```
Comment (JeanneGrenet) [inline src/components/map-location.tsx:19]: We should use que `<MapPin /> `icon from lucid react to have the same icons everywhere

File: src/content/components/map-location.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+# Default
+
+<Demo>
+  <MapLocation
+    detail={{
+      title: "Place",
+      description: "Entire address...",
+    }}
+    hidePinIcon
```
Comment (JeanneGrenet) [inline src/content/components/map-location.mdx:16]: What is this props for ?

**Review by JeanneGrenet — CHANGES_REQUESTED** [review]

File: src/content/components/map-location.mdx (hunk 4 above)
Comment (IounOm) [reply src/content/components/map-location.mdx:16]: I see in meetup page that not have pin icon on it so I write this use for reuse between 2 page form design form this capture image. Component will show pin icon by default but can remove it by add props `hidePinIcon`

if pin icon always needs to be included. I'll remove this prop and set the icon as a fixed value instead.

Screenshot 2567-10-29 at 12 27 39
Screenshot 2567-10-29 at 12 28 33

Comment (IounOm) [reply src/content/components/map-location.mdx]: fixed!

File: src/components/map-location.tsx (hunk 3 above)
Comment (IounOm) [reply src/components/map-location.tsx]: fixed!

**Review by houssembaltii — APPROVED** [review]

**Review by JeanneGrenet — APPROVED** [review]

**Pull request merged**